packages/toolkit/src/autoBatchEnhancer.ts

```diff
@@ -66,7 +59,8 @@ export const autoBatchEnhancer =
       options.type === 'tick'
         ? queueMicrotask
         : options.type === 'raf'
-          ? rAF
+          ? // requestAnimationFrame won't exist in SSR environments. Fall back to a vague approximation just to keep from erroring.
+            window?.requestAnimationFrame ?? createQueueWithTimer(10)
```

Comment:
`window?` doesn't cover all cases that `typeof window !== 'undefined'` does - optional chaining still can cause a reference error if a variable with that name never was defined:
![grafik](https://github.com/user-attachments/assets/57344b6c-4d91-45b9-ba86-9fed019ca5d9)

Reply:
ah, of course. I have just pushed a fix to revert it back to a ternary like before